packages/esm-amd-loader/src/esm-amd-loader.ts

```diff
@@ -278,6 +289,7 @@ function loadDeps(
 
     // We have a dependency on a real module.
     const dependency = getModule(resolveUrl(module.urlBase, depSpec));
+    dependency.crossorigin = module.crossorigin;
```

Comment:
We should only set crossorigin once, just to make things more predictable.

We could use `undefined` to mean, `not yet set to any value` and check for it here.

It'd still be racy, but I'm not sure how we could ameliorate that, as we want to do as much loading as we can as early as we can. In any case, it's a pretty obscure edge case, where you import the same dep both from a script with a crossorigin and from one without.